pkg/controllers/modeljob_controller_test.go

```diff
@@ -0,0 +1,199 @@
+	utilrand "k8s.io/apimachinery/pkg/util/rand"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	modeljobsv1alpha1 "github.com/kleveross/klever-model-registry/pkg/apis/modeljob/v1alpha1"
+	test "github.com/kleveross/klever-model-registry/testutil"
+)
+
+var _ = Describe("Modeljob Controller", func() {
+	initGlobalVar()
+	PresetAnalyzeImageConfig = test.InitPresetModelImageConfigMap()
+	modeljobName := ""
+
+	req := ctrl.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      modeljobName,
+			Namespace: "default",
+		},
+	}
+
+	const timeout = time.Second * 5
+	const interval = time.Second * 1
+
+	BeforeEach(func() {
+		modeljobName = fmt.Sprintf("modeljob-test-%s", utilrand.String(8))
+		// failed test runs that don't clean up leave resources behind.
+		modeljob := &modeljobsv1alpha1.ModelJob{}
+		err := k8sClient.DeleteAllOf(context.Background(), modeljob, client.InNamespace("default"))
+		Expect(err).ToNot(HaveOccurred())
+
+		job := &batchv1.Job{}
+		err = k8sClient.DeleteAllOf(context.Background(), job, client.InNamespace("default"))
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		// Add any teardown steps that needs to be executed after each test
+	})
+
+	Context("ModelJob", func() {
+		It("Should be created correctly", func() {
+			key := types.NamespacedName{
+				Name:      modeljobName,
+				Namespace: "default",
+			}
+
+			toCreate := &modeljobsv1alpha1.ModelJob{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      key.Name,
+					Namespace: key.Namespace,
+				},
+				Spec: modeljobsv1alpha1.ModelJobSpec{
+					Model: "extraction model tag",
+					ModelJobSource: modeljobsv1alpha1.ModelJobSource{
+						Extraction: &modeljobsv1alpha1.ExtractionSource{
+							Format: modeljobsv1alpha1.FormatSavedModel,
+						},
+					},
+				},
+			}
+			Expect(k8sClient.Create(context.Background(), toCreate)).Should(Succeed())
+
+			By("Expecting to reconcile a job successfully, the result is get job successfully")
+			Eventually(func() error {
+				getedModelJob := &modeljobsv1alpha1.ModelJob{}
+				err := k8sClient.Get(context.Background(), key, getedModelJob)
+				if err != nil {
+					return err
+				}
+
+				reconciler.Reconcile(req)
+				job := &batchv1.Job{}
+				return k8sClient.Get(context.Background(), key, job)
+			}, timeout, interval).Should(Succeed())
+
+			By("Expecting to reconcile a job successfully, the modejob status update to active")
+			Eventually(func() error {
+				job := &batchv1.Job{}
+				err := k8sClient.Get(context.Background(), key, job)
+				Expect(err).To(BeNil())
+
+				job.Status.StartTime = &metav1.Time{
+					Time: time.Now(),
+				}
+				job.Status.Active = 1
+				err = k8sClient.Status().Update(context.Background(), job)
+				if err != nil {
+					return err
+				}
+
+				_, err = reconciler.Reconcile(req)
+				if err != nil {
+					return err
+				}
+
+				job = &batchv1.Job{}
+				return k8sClient.Get(context.Background(), key, job)
```

Comment:
I think we should check more here. If we only check if we can get the job, there is no need to update the status.

Reply:
Yes, I will optimize some logic next